xarrayutils/utils.py

```diff
@@ -887,3 +887,40 @@ def concat_dim_da(data, name):
     return xr.DataArray(data, dims=[name],
                         coords={name: (name, data)},
                         name=name)
+
+
+
+def da_detrend(a, b, start_shift=0, timedim='time'):
+    """Detrends the dataarray b using slope from xr_linregress. start_shift 
+value is necessary to account for different time periods                    
+    e.g. when a forced run is coupled out at a certain point.               
+    Positive values indicate that the trend is starting earlier than the dat
+a, and vice versa for negative values.                                      
+    Time is converted from datetime to float so that unequally spaced timest
+eps can be handled (vs. using integer indexing)                             
+    Time dimension should have no chunks, i.e. v = v.chunk({'time': -1})    
+    a : {xr.DataArray}                                                      
+        Time variable for linear regression in datetime format              
+        before calling xr_linregress. This should be ok regardless of input 
+a.                                                                          
+                                                                            
+    b : {xr.DataArray, xr.Dataset}                                          
+        Dependent variable.                                                 
+    start_shift: datetime format                                            
```

Comment:
Do we actually need the start_shift when we calculate the trend using the offset? I assumed this was just a crux from my previous implementation.

Reply:
Not sure why it would be needed